llama-index-core/llama_index/core/base/embeddings/base.py

```diff
@@ -63,27 +63,28 @@ def similarity(
 class BaseEmbedding(TransformComponent, DispatcherSpanMixin):
     """Base class for embeddings."""
 
+    model_config = ConfigDict(
+        protected_namespaces=("pydantic_model_",), arbitrary_types_allowed=True
+    )
     model_name: str = Field(
         default="unknown", description="The name of the embedding model."
     )
     embed_batch_size: int = Field(
         default=DEFAULT_EMBED_BATCH_SIZE,
         description="The batch size for embedding calls.",
         gt=0,
-        lte=2048,
+        le=2048,
     )
-    callback_manager: CallbackManager = Field(
+    callback_manager: Optional[CallbackManager] = Field(
```

Comment:
Yeah, sorry I must've done that when I was first understanding how to validate/serialize non-BaseModel types Pydantic V2.

I have since done so using a better way, and now this can go back to not being Optional. I have made `CallbackManager` as non Optional whenever used as a Field in a BaseModel.

https://github.com/run-llama/llama_index/pull/15282/commits/64a4ea9a6aadd958c2f36d08ccb986968e47b5df